File: pythonVersion/company.py
class Company():
    def __init__(self, name, share_value, forecast, rate_of_change, forecast_influence):
        self.name = name
        self.share_value = share_value
        self.forecast = forecast
        self.rate_of_change = rate_of_change
        self.forecast_influence = forecast_influence
        
        self.share_value_history = [share_value]
        self.inventory = {}

    def buy(investor, amount):
        price = inventory[0].key() * amount
        approved_transaction = investor.withdraw(price)

File: cpp_variant/company.cpp
#include "company.h"

Company::Company(std::string name, int shareValue, std::pair<int, int> forecast, std::pair<float, float> rateOfChange, int forecastInfluence)
    : name(name), shareValue(shareValue), forecast(forecast), rateOfChange(rateOfChange), forecastInfluence(forecastInfluence) {
    shareValueHistory.push_back(shareValue);
}
std::string Company::getName() {
    return name;
}
float Company::getShareValue() {
    return shareValue;
}
std::vector<float> Company::getShareValueHistory() {
    return shareValueHistory;
}
std::pair<int, int> Company::getForecast() {
    return forecast;
}
int Company::getForecastInfluence() {
    return forecastInfluence;
}
std::pair<float, float> Company::getRateOfChange() {
    return rateOfChange;
}
std::map<double, Stock> Company::getInventory() {
    return inventory;
}
void Company::buy(Investor& investor, int amount) {
    double price = inventory.begin()->first * amount;
    bool approvedTransaction = investor.withdraw(price);

    Stock newStock(amount, companyID);

    investor.recieveStock();
}